x/mint/simulation/genesis.go

```diff
@@ -5,46 +5,199 @@ package simulation
+		func(r *rand.Rand) { reductionPeriodInEpochs = genReductionPeriodInEpochs(r) },
+	)
+
+	var distributionProportions types.DistributionProportions
+	simState.AppParams.GetOrGenerate(
+		simState.Cdc, distributionProportionsKey, &distributionProportions, simState.Rand,
+		func(r *rand.Rand) { distributionProportions = genDistributionProportions(r) },
+	)
 
-	// var maxRewardPerEpoch sdk.Dec
-	// simState.AppParams.GetOrGenerate(
-	// 	simState.Cdc, MaxRewardPerEpoch, &maxRewardPerEpoch, simState.Rand,
-	// 	func(r *rand.Rand) { maxRewardPerEpoch = GenMaxRewardPerEpoch(r) },
-	// )
+	var weightedDevRewardReceivers []types.WeightedAddress
+	simState.AppParams.GetOrGenerate(
+		simState.Cdc, weightedDevRewardReceiversKey, &weightedDevRewardReceivers, simState.Rand,
+		func(r *rand.Rand) { weightedDevRewardReceivers = genWeightedDevRewardReceivers(simState.Rand) },
+	)
 
-	// var minRewardPerEpoch sdk.Dec
-	// simState.AppParams.GetOrGenerate(
-	// 	simState.Cdc, MinRewardPerEpoch, &minRewardPerEpoch, simState.Rand,
-	// 	func(r *rand.Rand) { minRewardPerEpoch = GenMinRewardPerEpoch(r) },
-	// )
-	// Leaving as sample code
+	var mintintRewardsDistributionStartEpoch int64
+	simState.AppParams.GetOrGenerate(
+		simState.Cdc, mintingRewardsDistributionStartEpochKey, &mintintRewardsDistributionStartEpoch, simState.Rand,
+		func(r *rand.Rand) { mintintRewardsDistributionStartEpoch = genMintintRewardsDistributionStartEpoch(r) },
+	)
+
+	reductionStartedEpoch := genReductionStartedEpoch(simState.Rand)
 
 	mintDenom := sdk.DefaultBondDenom
-	epochProvisions := sdk.NewDec(500000) // TODO: Randomize this
-	params := types.NewParams(mintDenom, epochProvisions, "week", sdk.NewDecWithPrec(5, 1), 156, types.DistributionProportions{
-		Staking:          sdk.NewDecWithPrec(4, 1), // 0.4
-		PoolIncentives:   sdk.NewDecWithPrec(3, 1), // 0.3
-		DeveloperRewards: sdk.NewDecWithPrec(2, 1), // 0.2
-		CommunityPool:    sdk.NewDecWithPrec(1, 1), // 0.1
-	}, []types.WeightedAddress{}, 0)
+	params := types.NewParams(
+		mintDenom,
+		epochProvisions,
+		epochIdentifier,
+		reductionFactor,
+		reductionPeriodInEpochs,
+		distributionProportions,
+		weightedDevRewardReceivers,
+		mintintRewardsDistributionStartEpoch)
+
+	minter := types.NewMinter(epochProvisions)
 
-	mintGenesis := types.NewGenesisState(types.InitialMinter(), params, 0)
+	mintGenesis := types.NewGenesisState(minter, params, reductionStartedEpoch)
 
 	bz, err := json.MarshalIndent(&mintGenesis, "", " ")
 	if err != nil {
 		panic(err)
 	}
-	// TODO: Do some randomization later
-	fmt.Printf("Selected deterministically generated minting parameters:\n%s\n", bz)
+	fmt.Printf("Selected pseudo-randomly generated minting parameters:\n%s\n", bz)
 	simState.GenState[types.ModuleName] = simState.Cdc.MustMarshalJSON(mintGenesis)
 }
+
+func genEpochProvisions(r *rand.Rand) sdk.Dec {
+	return sdk.NewDec(int64(r.Intn(maxInt64)))
+}
+
+func genEpochIdentifier(r *rand.Rand) string {
+	return "day"
```

Comment:
To investigate whether this is expected: choosing between "day" and "week" here causes `TestAppStateDeterminism` to fail